### PR TITLE
Serve virtctl binaries from cluster for disconnected environments

### PR DESCRIPTION
**Pull request by erkanerol**
Now, we are redirecting users to github.com download page
when they want to download virtctl binary. However, it doesn't
work in disconnected environment. Therefore, we are starting
serving the binaries from the cluster so that users can download it
in disconnected environments.

Signed-off-by: Erkan Erol <eerol@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
As in #1466 - I think the name `downloads` is too generic, and also very different than the image name `virt-artifacts`. Please consider changing it.

File: deploy/index-image/community-kubevirt-hyperconverged/1.5.0/manifests/kubevirt-hyperconverged-operator.v1.5.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -2001,6 +2013,40 @@ spec:
                     memory: 48Mi
               priorityClassName: system-node-critical
               serviceAccountName: hyperconverged-cluster-operator
+      - label:
+          app.kubernetes.io/component: deployment
+          app.kubernetes.io/managed-by: olm
+          app.kubernetes.io/part-of: hyperconverged-cluster
+          app.kubernetes.io/version: 1.5.0
+          name: downloads
+        name: downloads
```
Comment (tiraboschi) [inline deploy/index-image/community-kubevirt-hyperconverged/1.5.0/manifests/kubevirt-hyperconverged-operator.v1.5.0.clusterserviceversion.yaml:2022]: hyperconverged-cluster-cli-download

File: pkg/components/components.go [hunk 2]
```diff
@@ -40,13 +40,21 @@ const (
 	hcoWhDeploymentName = "hco-webhook"
 	certVolume          = "apiservice-cert"
 
+	downloadsName = "downloads"
```
Comment (tiraboschi) [inline pkg/components/components.go:43]: hyperconverged-cluster-cli-download

File: pkg/controller/operands/cliDownload.go [hunk 3]
```diff
@@ -2,20 +2,29 @@ package operands
...
 	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/common"
 	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
 	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
 	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	downloadsServiceName = "downloads"
```
Comment (tiraboschi) [inline pkg/controller/operands/cliDownload.go:22]: hyperconverged-cluster-cli-downloads

**Comment by erkanerol** [comment]
> Note: The name of the new container (`virt-artifacts-server`) is different than the naming (`hyperconverged-cluster-cli-download`)  in our repo since we had some build pipelines for this new image and this image belongs to kubevirt side there and we followed their naming convention.

**Review by tiraboschi — APPROVED** [review]

**Comment by tiraboschi** [comment]
labels are not set and hco-e2e-image-index-gcp and hco-e2e-kv-smoke-gcp lanes are not overridden just due to kubevirt-ci outage.
Manually merging.

**Pull request merged**